nexus_constructor/stream_fields_widget.py

```diff
@@ -335,44 +336,39 @@ def get_stream_group(self) -> StreamGroup:
 
         return stream_group
 
-    def _create_ev42_fields(self, stream_group: EV42Stream):
+    def _record_advanced_f142_values(self, stream: F142Stream):
         """
-        Create ev42 fields in the given group if advanced options are specified.
-        :param stream_group: The group to apply fields to.
+        Save the advanced f142 properties to the stream data object.
+        :param stream: The stream data object to be modified.
         """
-        raise NotImplementedError
-        if self.advanced_options_enabled:
-            if self.ev42_adc_pulse_debug_checkbox.isChecked():
-                stream_group.create_dataset(
-                    ADC_PULSE_DEBUG,
-                    dtype=bool,
-                    data=self.ev42_adc_pulse_debug_checkbox.isChecked(),
-                )
-            self._create_dataset_from_spinner(
-                stream_group, self.ev42_nexus_to_spinner_ui_element
-            )
-
-    def _create_f142_fields(self, stream_group: F142Stream):
+        stream.index_every_mb = self.f142_nexus_to_spinner_ui_element[
+            NEXUS_INDICES_INDEX_EVERY_MB
+        ].value()
+        stream.index_every_kb = self.f142_nexus_to_spinner_ui_element[
+            NEXUS_INDICES_INDEX_EVERY_KB
```

Comment:
I wonder if we should just scrap the `f142_nexus_to_spinner_ui_element` dict and use the value from the UI element in `self` here. it doesn't really make things look neat with the new model change and would be much clearer to just call something like this: 
`stream.index_every_mb = self.index_every_mb_spinbox.value()`

Reply:
I agree. Alternatively, you could stay closer to how this was implemented before by using setattr:
```python
setattr(stream, NEXUS_INDICES_INDEX_EVERY_MB.split('.')[-1], self.f142_nexus_to_spinner_ui_element[NEXUS_INDICES_INDEX_EVERY_MB].value())
```
which would allow a more similar approach to before with looping through the dictionary.